internal/cmd/iam/command_provider_update.go

```diff
@@ -58,7 +59,14 @@ func (c *identityProviderCommand) update(cmd *cobra.Command, args []string) erro
 
 	identityProvider, err := c.V2Client.UpdateIdentityProvider(update)
 	if err != nil {
-		return err
+		provId, err := pconv.ConvertIamProviderNameToId(args[0], c.V2Client)
```

Comment:
I would call this variable `providerId` or just `id`, not a huge fan of shortening names unless it's more common (for example, "directory" -> "dir"). No one looks at "prov" and immediately thinks "provider"